logging/resource.go

```diff
@@ -34,165 +33,172 @@ type commonResource struct{ *mrpb.MonitoredResource }
 
 func (r commonResource) set(l *Logger) { l.commonResource = r.MonitoredResource }
 
-var detectedResource struct {
-	pb   *mrpb.MonitoredResource
-	once sync.Once
+var detectedResource = struct {
+	pb    *mrpb.MonitoredResource
+	attrs internal.ResourceAtttributesGetter
+	once  *sync.Once
+}{
+	attrs: internal.ResourceAttributes(),
+	once:  new(sync.Once),
+}
+
+func metadataProjectID() (string, bool) {
+	return detectedResource.attrs.Metadata("project/project-id")
+}
+
+func metadataZone() (string, bool) {
```

Comment:
nit: all of these functions that use detectedResource feel like they should instead be methods on that type. Having functions work on global